spec/models/bottleneck_event_spec.rb

```diff
@@ -107,4 +107,41 @@
       it { is_expected.to be_same_time_as(bottleneck_event.created_on) }
     end
   end
+
+  describe ".delete_future_events_for_obj" do
+    let(:resource) { FactoryGirl.create(resource_name) }
+    let!(:bottleneck_event) { BottleneckEvent.create!(:resource => resource, :future => true) }
+
+    context "for a host_redhat resource" do
+      let(:resource_name) { :host_redhat }
+      it "deletes the future event" do
+        BottleneckEvent.delete_future_events_for_obj(resource)
```

Comment:
You can remove this line (and all matching lines in the below examples) and do this once within the outside `describe` block:

```
before { BottleneckEvent.delete_future_events_for_obj(resource) }

context "for a host_redhat resource" do
...
...
```